pyproject.toml

```diff
@@ -35,6 +35,7 @@ classifiers = [
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
+    "dask",
```

Comment:
Is it required as dependency? The code runs fine without dask right?

Reply:
Good point, I'll check it without and see if it still works.

Reply:
I added dask back in as the tests broke on github actions after removing, so perhaps it is needed.